### ISSUE TITLE
Tilt should recognize all clusters with 'microk8s' cluster name prefix as microk8s

### ISSUE THREAD

**Issue by sirianni** [tilt-dev/tilt#4300]
### Expected Behavior
Tilt deploys to local microk8s cluster without complaining.

### Current Behavior
```
Stop! microk8s might be production.
If you're sure you want to deploy there, add:
	allow_k8s_contexts('microk8s')
```
#### `tilt doctor` Output
```
$ tilt doctor
Tilt: v0.18.11, built 2021-02-26
System: linux-amd64
---
Docker
- Host: [default]
- Version: 1.41
- Builder: 2
---
Kubernetes
- Env: unknown
- Context: microk8s
- Cluster Name: microk8s
- Namespace: default
- Container Runtime: containerd
- Version: v1.19.7-34+02d22c9f4fb254
- Cluster Local Registry: none
---
Thanks for seeing the Tilt Doctor!
Please send the info above when filing bug reports. 💗

The info below helps us understand how you're using Tilt so we can improve,
but is not required to ask for help.
---
Analytics Settings
--> (These results reflect your personal opt in/out status and may be overridden by an `analytics_settings` call in your Tiltfile)
- User Mode: opt-in
- Machine: 3b8f44a97f22912f6fc75e833d363d24
- Repo: dUlGzRW35jXfKwQEzY6/kQ==
```

This used to work fine previously, so I'm not sure what has changed.
This looks suspicious, however I am able to push to `localhost:32000` using `docker push`
```
- Cluster Local Registry: none
```

**Comment by nicks** [comment]
Thanks for the detailed report!

The doctor output is confusing to me. Tilt uses the cluster name to determine what cluster it's talking to. AFAIK, the microk8s cluster name should always be 'microk8s-cluster' (see: https://github.com/ubuntu/microk8s/blob/master/microk8s-resources/client.config#L5).

But it appears your cluster has the name 'microk8s', not 'microk8s-cluster'. Do you have any ideas on how you got into that state? (It's easy enough to fix, I'm just curious how this happened, and if it indicates to some other mistaken assumption)

**Comment by sirianni** [comment]
> But it appears your cluster has the name 'microk8s', not 'microk8s-cluster'. Do you have any ideas on how you got into that state?

Yes. I use this script to configure my Kubernetes context for microk8s
```
kubectl config set-cluster microk8s --server=https://127.0.0.1:16443/ --certificate-authority=/var/snap/microk8s/current/certs/ca.crt
kubectl config set-credentials microk8s-admin --token=$(microk8s kubectl config view --raw -o 'jsonpath={.users[0].user.token}')
kubectl config set-context microk8s --cluster=microk8s --namespace=default --user=microk8s-admin
```

The `microk8s config` command does configure the cluster name as `microk8s-cluster`, but using that command requires manually merging that config with your existing `~/.kube/config`, which is why I prefer the scripted approach.  It's easy enough to change my script to set the cluster name as `microk8s-cluster`, but may also be worth exploring a more robust way for `tilt` to detect microk8s.  For example, it looks like there is a `"microk8s.io/cluster": "true"` label on the k8s node object:

```
$ kubectl get nodes -o 'jsonpath={.items[0].metadata.labels}' | jq
```
```json
{
  "beta.kubernetes.io/arch": "amd64",
  "beta.kubernetes.io/os": "linux",
  "kubernetes.io/arch": "amd64",
  "kubernetes.io/hostname": "sirianni-tp",
  "kubernetes.io/os": "linux",
  "microk8s.io/cluster": "true"
}
```

**Comment by nicks** [comment]
ah, ok.

Sounds like an easy workaround is to change the cluster name to microk8s-cluster in your script. That will work with existing versions of Tilt.

I'd also be fine with a PR to change tilt to treat any cluster with the prefix 'microk8s' as a microk8s cluster.

Re: more robust ways to detect cluster type - ya, I had a discussion a while back with SIG-cluster-lifecycle about ways to communicate this sort of information. node annotations were one of the options on the table. It has a bunch of downsides and they dismissed it pretty quickly. Like it or not, cluster-name is the mechanism that most of the cluster maintainers we work with have agreed on, because it's fairly robust, simple to implement, and simple to correct.
